File: frontend/types/sync.ts
export interface SyncManualResponse {
  last_sync: string;
  tablas_procesadas: string[];
  registros_aplicados: number;
}

export interface SyncTableStatus {
  tabla: string;
  pendientes_postgres: number;
  pendientes_mysql: number;
}

export interface SyncStatusResponse {
  last_sync_postgres: string | null;
  last_sync_mysql: string | null;
  effective_last_sync: string | null;
  tablas: SyncTableStatus[];
}

File: backend/app/schemas/sync.py
from datetime import datetime

from pydantic import BaseModel


class SyncManualResponse(BaseModel):
    last_sync: datetime
    tablas_procesadas: list[str]
    registros_aplicados: int


class SyncTableStatus(BaseModel):
    tabla: str
    pendientes_postgres: int
    pendientes_mysql: int


class SyncStatusResponse(BaseModel):
    last_sync_postgres: datetime | None
    last_sync_mysql: datetime | None
    effective_last_sync: datetime | None
    tablas: list[SyncTableStatus]
